graylog2-server/src/main/java/org/graylog/plugins/sidecar/rest/resources/ConfigurationResource.java

```diff
@@ -332,6 +339,50 @@ public Response deleteConfiguration(@ApiParam(name = "id", required = true)
         return Response.accepted().build();
     }
 
+    private ValidationResult validate(Configuration toValidate) {
+        final Optional<Configuration> configurationOptional;
+        final Configuration configuration;
+        final ValidationResult validation = new ValidationResult();
+
+        configurationOptional = Optional.ofNullable(configurationService.findByName(toValidate.name()));
+        if (configurationOptional.isPresent()) {
+            configuration = configurationOptional.get();
+            if (!configuration.id().equals(toValidate.id())) {
+                // a configuration exists with a different id, so the name is already in use, fail validation
+                validation.addError("name", "Configuration \"" + toValidate.name() + "\" already exists");
+            }
+        }
+
+        if (toValidate.name().isEmpty()) {
+            validation.addError("name", "Configuration name cannot be empty.");
+        } else if (!VALID_NAME_PATTERN.matcher(toValidate.name()).matches()) {
+                validation.addError("name", "Configuration name can not include the following characters: ; * ? \" < > | &");
+        }
+
+        if (toValidate.collectorId().isEmpty()) {
+            validation.addError("collector_id", "Associated collector ID cannot be empty.");
+        }
+
+        if (toValidate.color().isEmpty()) {
+            validation.addError("color", "Collector color cannot be empty.");
+        }
+
+        if (toValidate.template().isEmpty()) {
```

Comment:
This does not trim the input, so adding a space to it would make the validation fail. I guess that should be done in all required fields.

Reply:
We can't use trim here as a single whitespace is used as default value for incomplete object validations.